File: catkin_ws/src/my_package/scripts/topic_subscriber.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int32

def feedback_cb(msg):
	print(msg.data)


def main():
	rospy.init_node('topic_subscriber')
	sub = rospy.Subscriber('/counter', Int32, feedback_cb)
	rospy.spin()

if __name__ == '__main__':
	main()

File: catkin_ws/src/my_package/src/topic_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include <iostream>

void FeedbackCB(const std_msgs::Int32::ConstPtr& msg)
{
	ROS_INFO("Count is: [%d]", msg->data);
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "topic_subscriber");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/counter", 10, FeedbackCB);
	ros::spin();
	return 0;
}